VultisigApp/VultisigApp/Services/Solana/SolanaService.swift

```diff
@@ -26,29 +33,35 @@ class SolanaService {
             return response.result
             
         } catch {
-            
-            
-            
             print("Error sending transaction: \(error.localizedDescription)")
         }
         return nil
     }
     
-    func getSolanaBalance(coin: Coin) async throws -> (rawBalance: String, priceRate: Double){
+    func getSolanaBalance(coin: Coin) async throws -> (rawBalance: String, priceRate: Double) {
         var rawBalance = "0"
         let priceRateFiat = await CryptoPriceService.shared.getPrice(priceProviderId: coin.priceProviderId)
         
         do {
-            let data = try await Utils.PostRequestRpc(rpcURL: rpcURL, method: "getBalance", params:  [coin.address])
             
-            if let totalBalance = Utils.extractResultFromJson(fromData: data, path: "result.value") as? Int64 {
-                rawBalance = totalBalance.description
-            }            
+            if coin.isNativeToken {
+                
+                let data = try await Utils.PostRequestRpc(rpcURL: rpcURL, method: "getBalance", params: [coin.address])
```

Comment:
use Codable and JSONDecoder

Reply:
I've personally remove this Utils.PostRequestRpc, to not work with raw data anymore

Reply:
I don't think it's worth the level of effort to keep strong typing everything. In this case, we need one property only. It is a base type, so I would need to create a structure to map one item. Well, we can do it, but ...